src/Common/tests/UseDefaultConnectionLimitsAttribute.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Net;
+using System.Reflection;
+using Xunit.Sdk;
+
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+public class UseEndpointConnectionLimitAttribute : BeforeAfterTestAttribute
```

Comment:
This attribute can't be used on more than one test in the project. Think of the scenario where TestA uses this attribute and is running. Just before TestA ends, TestB also using this attribute starts. Then TestA ends and resets the connection limit just before any client connections are created in TestB, causing TestB to fail.  
This can be solved by adding a static synchronization primitive, eg a SemaphoreSlim with a capacity of 1. In the Before method, call Wait() with a high enough timeout to allow for another test (or even multiple) to complete, and in the After method, call Release().

Reply:
done